DS_and_Algorithm/Selenium/test.java

```diff
@@ -0,0 +1,706 @@
+    driver.get(pages.formPage);
+    assertCanToggle(enabledUnselectedCheckbox());
+  }
+
+  @Test
+  void testShouldBeAbleToToggleAnEnabledSelectedCheckbox() {
+    driver.get(pages.formPage);
+    assertCanToggle(enabledSelectedCheckbox());
+  }
+
+  @Test
+  void testClickingOnASelectedRadioButtonShouldLeaveItSelected() {
+    driver.get(pages.formPage);
+
+    WebElement button = enabledSelectedRadioButton();
+    assertThat(button.isSelected()).isTrue();
+
+    button.click();
+
+    assertThat(button.isSelected()).isTrue();
+  }
+
+  @Test
+  void testShouldBeAbleToToggleEnabledMultiSelectOption() {
+    driver.get(pages.formPage);
+    assertCanToggle(selectedMultipleSelectOption());
+  }
+
+  @Test
+  void testShouldBeAbleToToggleSelectableCheckboxByClickingOnIt() {
+    driver.get(pages.formPage);
+
+    WebElement checkbox = enabledUnselectedCheckbox();
+    assertNotSelected(checkbox);
+
+    checkbox.click();
+    assertSelected(checkbox);
+
+    checkbox.click();
+    assertNotSelected(checkbox);
+  }
+
+  @Test
+  void testShouldBeAbleToSelectSelectableRadioButtonByClickingOnIt() {
+    driver.get(pages.formPage);
+
+    WebElement radioButton = enabledUnselectedRadioButton();
+    assertNotSelected(radioButton);
+
+    radioButton.click();
+    assertSelected(radioButton);
+
+    radioButton.click();
+    assertSelected(radioButton);
+  }
+
+  @Test
+  void testClickingDisabledSelectedCheckboxShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(randomlyDisabledSelectedCheckbox());
+  }
+
+  @Test
+  void testClickingDisabledUnselectedCheckboxShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledUnselectedCheckbox());
+  }
+
+  @Test
+  void testClickingDisabledSelectedRadioButtonShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledSelectedRadioButton());
+  }
+
+  @Test
+  void testClickingDisabledUnselectedRadioButtonShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledUnselectedRadioButton());
+  }
+
+  private void assertNotSelected(WebElement element) {
+    assertSelected(element, UNSELECTED);
+  }
+
+  private void assertSelected(WebElement element) {
+    assertSelected(element, SELECTED);
+  }
+
+  private void assertSelected(WebElement element, boolean isSelected) {
+    wait.until(ExpectedConditions.elementSelectionStateToBe(element, isSelected));
+    assertThat(element.isSelected())
+        .describedAs(
+            "Expected element %s to be %s",
+            describe(element), selectedToString(isSelected), selectedToString(!isSelected))
+        .isEqualTo(isSelected);
+  }
+
+  private void assertCannotSelect(WebElement element) {
+    boolean previous = element.isSelected();
+    element.click();
+    assertThat(element.isSelected()).isEqualTo(previous);
+  }
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/16#issuecomment-2222577709>#69eb55</a><span> - 07/11/2024, 10:37 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The function 'assertCannotSelect' does not handle potential exceptions that may arise from the 'element.click()' method, such as 'ElementNotInteractableException'.
</div>
<div id="fix">
<b>Fix:</b> Add a try-catch block to handle potential exceptions from the 'element.click()' method.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -398,7 +398,11 @@
     private void assertCannotSelect(WebElement element) {
         boolean previous = element.isSelected();
 -        element.click();
 +        try {
 +            element.click();
 +        } catch (ElementNotInteractableException e) {
 +            // Handle exception or log it
 +        }
         assertThat(element.isSelected()).isEqualTo(previous);
     }
```
</div>
</details>
</div>
<br>


</div>